packages/snjs/lib/services/history/history_manager.ts

```diff
@@ -226,24 +317,95 @@ export class SNHistoryManager extends PureService {
    * Expands on a revision fetched via `remoteHistoryForItem` by getting a revision's
    * complete fields (including encrypted content).
    */
-  async fetchRemoteRevision(itemUuid: UuidString, entry: RevisionListEntry) {
-    const revision = (await this.apiService!.getRevision(
+  async fetchRemoteRevision(
+    itemUuid: UuidString,
+    entry: RevisionListEntry
+  ): Promise<HistoryEntry | undefined> {
+    const revision = (await this.apiService.getRevision(
       entry,
       itemUuid
     )) as SingleRevision;
     if ((revision as SingleRevisionResponse).error) {
       return undefined;
     }
-    const payload = CreateMaxPayloadFromAnyObject(revision as any, {
-      uuid: revision.item_uuid,
-    });
+    const payload = CreateMaxPayloadFromAnyObject(
+      (revision as unknown) as RawPayload,
+      {
+        uuid: revision.item_uuid,
+      }
+    );
     const encryptedPayload = CreateSourcedPayloadFromObject(
       payload,
       PayloadSource.RemoteHistory
     );
-    const decryptedPayload = await this.protocolService!.payloadByDecryptingPayload(
+    const decryptedPayload = await this.protocolService.payloadByDecryptingPayload(
       encryptedPayload
     );
-    return new ItemHistoryEntry(decryptedPayload);
+    if (decryptedPayload.errorDecrypting) {
+      return undefined;
+    }
+    return new HistoryEntry(decryptedPayload as SurePayload);
+  }
+
+  optimizeHistoryForItem(uuid: string): void {
+    /**
+     * Clean up if there are too many revisions. Note itemRevisionThreshold
+     * is the amount of revisions which above, call for an optimization. An
+     * optimization may not remove entries above this threshold. It will
+     * determine what it should keep and what it shouldn't. So, it is possible
+     * to have a threshold of 60 but have 600 entries, if the item history deems
+     * those worth keeping.
+     *
+     * Rules:
+     * - Keep an entry if it is the oldest entry
+     * - Keep an entry if it is the latest entry
+     * - Keep an entry if it is Significant
+     * - If an entry is Significant and it is a deletion change, keep the entry before this entry.
+     */
```

Comment:
This comment should probably be above the function instead of inside its body, no?